### ISSUE TITLE
ratpack always response 2 Reassembled TCP Segments to client

### ISSUE THREAD

**Issue by tianmaogen** [ratpack/ratpack#1328]
Hi.
I use ratpack to replace `netty` in my project currently, and find that using ratpack always response 2 `Reassembled TCP Segments` to client.

> Bellow is the network packages by wireshark:

![1516351279](https://user-images.githubusercontent.com/9186391/35141974-cff1da9a-fd37-11e7-8017-33572e240468.png)

And i notice the bellow code in [ratpack.server.internal.DefaultResponseTransmitter](https://github.com/ratpack/ratpack/blob/master/ratpack-core/src/main/java/ratpack/server/internal/DefaultResponseTransmitter.java) shows how `ratpack` send 2 frames in <1>, <2>:
```
    private void transmit(HttpResponseStatus responseStatus, Object body, boolean sendLastHttpContent) {
        ChannelFuture channelFuture = this.pre(responseStatus);   <1>
        if(channelFuture == null) {
            ReferenceCountUtil.release(body);
            this.isKeepAlive = false;
            this.post(responseStatus);
        } else {
            channelFuture.addListener((future) -> {
                if(future.isSuccess() && this.channel.isOpen()) {
                    if(sendLastHttpContent) {
                        this.channel.write(body);
                        this.post(responseStatus);
                    } else {
                        this.post(responseStatus, this.channel.writeAndFlush(body));    <2>
                    }
                } else {
                    ReferenceCountUtil.release(body);
                    this.isKeepAlive = false;
                    this.post(responseStatus);
                }

            });
        }
    }
```

I want to ask is there another `xxxResponseTransmitter` to replace `DefaultResponseTransmitter` sending just one segment to client?

Thanks a lot for advice.

**Comment by ldaley** [comment]
Hi @tianmaogen,

There's no built in way to avoid the two segments right now. We can certainly change that if necessary. We flush the headers immediately so that the client can act as soon as possible on the response, based on those headers.

Can you please describe what problem this is causing you.

**Comment by tianmaogen** [comment]
Hi, @ldaley 
Thanks for so quickly response.
In our company, many old version of client use upstream module to fetch data from server, which can't support HTTP subcontracting. It's hard to change the client's old version code.

Can you please tell me do you have a plan to support for just one segment?

**Comment by ldaley** [comment]
@tianmaogen I've made this change speculatively for 1.6, but I will need to do some wider testing before committing to including it in the final version.

It would help if you could verify whether this actually fixes your problem or not. There's a chance that something lower in the stack than Ratpack still segments the transmission.

**Comment by tianmaogen** [comment]
@ldaley 
It's my pleasure to verify whether 1.6 actually fixes our problem.
Thank you very much!

**Comment by tianmaogen** [comment]
Hi, @ldaley 
I'm sorry to interrupt you again. For our customer requirement, our project have a deadline.
Can you please tell me the roughly time plan for `release-1.6.0`?

Thanks.

**Comment by ldaley** [comment]
@tianmaogen I have included this change in 1.5.3 which I am releasing now.

**Comment by tianmaogen** [comment]
@ldaley 
I have verified and 1.5.3 fixes our problem.
Thank you again for so quickly response.
